### PR TITLE
[libcaer,dv-processing] Port fixes, cleanup

### PR DESCRIPTION
**Pull request by dg0yt**
[libcaer] Fix export of libusb dependency.
[dv-processing] Fixes configuration race. (https://github.com/microsoft/vcpkg/pull/29843#issuecomment-1458339685)

`dv-processing[tools]` is known to build on Windows (MSVC, tested in vcpkg CI) and on macOS (more recent toolchain than in vcpkg CI, tested locally).

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
x64-windows-static:
~~~
FAILED: utilities/tcpstat/dv-tcpstat.exe 
cmd.exe /C "cd . && D:\downloads\tools\cmake-3.25.1-windows\cmake-3.25.1-windows-i386\bin\cmake.exe -E vs_link_exe --intdir=utilities\tcpstat\CMakeFiles\dv-tcpstat.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100226~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1434~1.319\bin\Hostx64\x64\link.exe  @CMakeFiles\dv-tcpstat.rsp  /out:utilities\tcpstat\dv-tcpstat.exe /implib:utilities\tcpstat\dv-tcpstat.lib /pdb:utilities\tcpstat\dv-tcpstat.pdb /version:0.0 /machine:x64 /nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF   /subsystem:console  -DYNAMICBASE -NXCOMPAT -LARGEADDRESSAWARE -OPT:REF -OPT:ICF -GUARD:CF  && cd ."
LINK: command "C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1434~1.319\bin\Hostx64\x64\link.exe @CMakeFiles\dv-tcpstat.rsp /out:utilities\tcpstat\dv-tcpstat.exe /implib:utilities\tcpstat\dv-tcpstat.lib /pdb:utilities\tcpstat\dv-tcpstat.pdb /version:0.0 /machine:x64 /nologo /DEBUG /INCREMENTAL:NO /OPT:REF /OPT:ICF /subsystem:console -DYNAMICBASE -NXCOMPAT -LARGEADDRESSAWARE -OPT:REF -OPT:ICF -GUARD:CF /MANIFEST /MANIFESTFILE:utilities\tcpstat\dv-tcpstat.exe.manifest" failed (exit code 1319) with the following output:
qwindows.lib(qwindowscontext.cpp.obj) : error LNK2038: mismatch detected for '_COROUTINE_ABI': value '1' doesn't match value '2' in tcpstat.cpp.obj
qwindows.lib(qwindowstheme.cpp.obj) : error LNK2038: mismatch detected for '_COROUTINE_ABI': value '1' doesn't match value '2' in tcpstat.cpp.obj
Qt6Core.lib(qfactorycacheregistration.cpp.obj) : error LNK2038: mismatch detected for '_COROUTINE_ABI': value '1' doesn't match value '2' in tcpstat.cpp.obj
Qt6Core.lib(qlocale_win.cpp.obj) : error LNK2038: mismatch detected for '_COROUTINE_ABI': value '1' doesn't match value '2' in tcpstat.cpp.obj


   Creating library utilities\tcpstat\dv-tcpstat.lib and object utilities\tcpstat\dv-tcpstat.exp


utilities\tcpstat\dv-tcpstat.exe : fatal error LNK1319: 4 mismatches detected
~~~

@Neumann-A Do you have an idea if this can be resolved? This port uses `-std:c++20`. What does Qt6 use? (Pulled in here via opencv4.)

**Comment by Neumann-A** [comment]
Qt uses 17

**Comment by Neumann-A** [comment]
https://devblogs.microsoft.com/oldnewthing/20230111-00/?p=107694

**Comment by dg0yt** [comment]
CC @llongi

**Comment by llongi** [comment]
Thanks for the improvements!
They look good to me, only two notes:
1) is it really safe to change (libcaer) and remove (dv-processing) the vcpkg_cmake_config_fixup(...) calls?
2) in dv-processing `DISABLE_PARALLEL_CONFIGURE # writes to include/cmake/version.hpp` that comment should refer to "include/dv-processing/version.hpp" as the correct path.

I agree with removing pkg-config files in dv-processing, its support is indeed incomplete as most packages are abandoning it and don't even ship the right files anymore.
I've imported the -D_ALLOW_COROUTINE_ABI_MISMATCH fix into the main CMakeLists.txt for the next official release.

**Comment by dg0yt** [comment]
> 1. is it really safe to change (libcaer) and remove (dv-processing) the vcpkg_cmake_config_fixup(...) calls?

Yes. The default `PACKAGE_NAME` is `${PORT}`.  The default `CONFIG_PATH` is `share/${PORT}`.

>  2. in dv-processing `DISABLE_PARALLEL_CONFIGURE # writes to include/cmake/version.hpp` that comment should refer to "include/dv-processing/version.hpp" as the correct path.

Indeed. But ideally, you would change upstream behaviour to not write to the source dir.

File: ports/dv-processing/vcpkg.json [hunk 1]
```diff
@@ -28,7 +40,11 @@
       "name": "vcpkg-cmake-config",
       "host": true
     },
```
Comment (LilyWangLL) [inline ports/dv-processing/vcpkg.json:42]:
```suggestion
```
This dependency should be removed.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/dv-processing/vcpkg.json (hunk 1 above)
Comment (dg0yt) [reply ports/dv-processing/vcpkg.json]: Done.

**Review by LilyWangLL — APPROVED** [review]

**Pull request merged**